common/sgx/iohandler/file_io.h

```diff
@@ -0,0 +1,44 @@
+/* Copyright 2019 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include <string>
+#include <stdint.h>
+
+class FileIoExecutor {
+public:
+    uint32_t GetIoHandlerId(const char* handlerName);
+
+    size_t GetMaxFileSize();
+
+    size_t GetMaxIoResultSize();
+
+    uint32_t FileOpen(uint32_t handler_id, std::string file_name, uint8_t *result,
```

Comment:
Do we need to pass handler_id and file_name in all class member functions? We can make handler_id and file_name as private members of class and initialize it in constructor of FileIoExecutor